src/Symfony/Component/Routing/Matcher/Dumper/PhpMatcherTrait.php

```diff
@@ -113,6 +109,14 @@ private function doMatch(string $pathinfo, array &$allow = [], array &$allowSche
                 }
                 continue;
             }
+
+            if ('/' !== $pathinfo && $hasTrailingSlash === ($trimmedPathinfo === $pathinfo)) {
```

Comment:
this should go before the previous "if" block

Reply:
Well, than the method-based variant of this bug will have to be accepted as a deliberate choice, won't it?

I.e. you will get a 404 on this:

```yaml
post.withTrail:
  path: /bar/ # GET /bar would become GET /bar/ due to this partial path-match
  method: [POST]

get.withoutTrail: # A request for GET /bar should match this route
  path: /bar # GET /bar/ does not match this path
  method: [GET]
```

When doing a GET-request on '/bar'

Reply:
> GET /bar would become GET /bar/ due to this partial path-match

not quite, since there is the `isset($requiredMethods['GET'])`, isn't it?

Reply:
Ah true, changed it :)

And the reverse variant hits the same issue as defining both a /foo and a /foo/ for GET anyway.